src/queries/enum_marked_non_exhaustive.ron

```diff
@@ -0,0 +1,57 @@
+SemverQuery(
+    id: "enum_marked_non_exhaustive",
+    human_readable_name: "enum marked #[non_exhaustive]",
+    description: "An exhaustive enum has been marked #[non_exhaustive].",
+    reference: Some("An exhaustive enum has been marked #[non_exhaustive], making any pattern-matching on it outside of its crate impossible until it includes a wildcard pattern like `_`."),
```

Comment:
Wording suggestion I just thought of, since `impossible` is a bit vague and we want to be specific about the consequence:
```suggestion
    reference: Some("An exhaustive enum has been marked #[non_exhaustive]. Pattern-matching on it outside of its crate must now include a wildcard pattern like `_`, or it will fail to compile."),
```